test/tools/fuzzTesting/fuzzHelper.cpp

```diff
@@ -273,91 +280,114 @@ void RandomCode::refreshSeed()
 	int pushCode = 96 + length - 1;
-	return toCompactHex(pushCode) + _hex;
+	return toCompactHex(pushCode) + hexVal;
 }
 
 std::string RandomCode::getPushCode(int _value)
 {
-	std::string hexString = toCompactHex(_value);
+	std::string hexString = toCompactHex(_value, HexPrefix::DontAdd, 1);
 	return getPushCode(hexString);
 }
 
-std::string RandomCode::fillArguments(dev::eth::Instruction _opcode, RandomCodeOptions const& _options)
+std::string RandomCode::fillArguments(eth::Instruction _opcode, RandomCodeOptions const& _options)
 {
-	dev::eth::InstructionInfo info = dev::eth::instructionInfo(_opcode);
+	eth::InstructionInfo info = eth::instructionInfo(_opcode);
 
 	std::string code;
 	bool smart = false;
-	unsigned num = info.args;
+	unsigned argsNum = info.args;
 	int rand = randUniIntGen() % 100;
 	if (rand < _options.smartCodeProbability)
 		smart = true;
 
 	if (smart)
 	{
 		//PUSH1 ... PUSH32
-		if (dev::eth::Instruction::PUSH1 <= _opcode && _opcode <= dev::eth::Instruction::PUSH32)
+		if (eth::Instruction::PUSH1 <= _opcode && _opcode <= eth::Instruction::PUSH32)
 		{
-		  code += rndByteSequence(int(_opcode) - int(dev::eth::Instruction::PUSH1) + 1);
-		  return code;
+			code += rndByteSequence(int(_opcode) - int(eth::Instruction::PUSH1) + 1);
+			return code;
 		}
 
 		//SWAP1 ... SWAP16 || DUP1 ... DUP16
-		bool isSWAP = (dev::eth::Instruction::SWAP1 <= _opcode && _opcode <= dev::eth::Instruction::SWAP16);
-		bool isDUP = (dev::eth::Instruction::DUP1 <= _opcode && _opcode <= dev::eth::Instruction::DUP16);
+		bool isSWAP = (eth::Instruction::SWAP1 <= _opcode && _opcode <= eth::Instruction::SWAP16);
+		bool isDUP = (eth::Instruction::DUP1 <= _opcode && _opcode <= eth::Instruction::DUP16);
 
 		if (isSWAP || isDUP)
 		{
 			int times = 0;
 			if (isSWAP)
-				times = int(_opcode) - int(dev::eth::Instruction::SWAP1) + 2;
+				times = int(_opcode) - int(eth::Instruction::SWAP1) + 2;
 			else
 			if (isDUP)
-				times = int(_opcode) - int(dev::eth::Instruction::DUP1) + 1;
+				times = int(_opcode) - int(eth::Instruction::DUP1) + 1;
 
 			for (int i = 0; i < times; i ++)
-				code += getPushCode(randUniIntGen() % 32);
+				code += getPushCode(rndByteSequence(randOpLengGen()));
 
 			return code;
 		}
 
 		switch (_opcode)
 		{
-		case dev::eth::Instruction::CREATE:
+		case eth::Instruction::EXTCODECOPY:
+			code += getPushCode(randUniIntGen());	//memstart2
+			code += getPushCode(randUniIntGen());	//memlen1
+			code += getPushCode(randUniIntGen());	//memstart1
+			code += getPushCode(toString(_options.getRandomAddress()));//address
+		break;
+		case eth::Instruction::EXTCODESIZE:
+			code += getPushCode(toString(_options.getRandomAddress()));//address
+		break;
+		case eth::Instruction::CREATE:
 			//(CREATE value mem1 mem2)
-			code += getPushCode(randUniIntGen() % 128);  //memlen1
-			code += getPushCode(randUniIntGen() % 32);   //memlen1
-			code += getPushCode(randUniIntGen());		 //value
+			code += getPushCode(randUniIntGen());	//memlen1
```

Comment:
Well, that means it will generate code accessing 1GB of memory on average for each instruction.

I guess the idea of generating addresses and lengths 0..127 and 0..32 was to produce the code closer to real life scenarios